current/advanced/block-import.md

```diff
@@ -0,0 +1,41 @@
+---
+slug: block-import
+lang: en
+title: The Block Import Pipeline
+---
+
+
+## The Import Queue
+
+The import queue is an abstract worker queue present in every Substrate node. It is not part of the runtime. The Import Queue is responsible for processing pieces of incoming information, verifying them, and if they are valid, importing them into the node's state. The most fundamental piece of information that the import queue processes is blocks themselves, but it is also responsible for importing consensus-related messages such as Justifications and, in light clients, finality proofs.
+
+The Import queue collects incoming blocks from the network or from other parts of the local node, and stores them in a pool. The blocks are later checked for validity and discarded if they are not valid. Finally, the block is imported when the import queue is instructed to import it by the consensus engine.
```

Comment:
The last phrase here is a bit problematic; 3 repetitions of "import" but mainly this notion of the consensus engine sending instructions to the queue is a bit surprising. Typically a queue is as dumb as you can get away with and designed to be decoupled from the logic responsible for processing its items. If the import queue is indeed different and "smarter" than any old queue I think a developer reading this would want a hint, something along the lines of "the import queue in substrate based blockchains is peculiar in that is bla bla and communicates with the consensus engine to decide if and when a block is to be imported".

Is the pool an in-memory store or is it persisted to disk to survive restarts? I think it's the latter but either way it's worth mentioning imo.

Reply:
Yeah, great point. I can re-phrase to avoid saying "import" too many times. But I'm not sure about the other points.

I thought it waited until the consensus engine instructed to do the importing based on the [rustdocs for the `ImportQueue` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.ImportQueue.html) They are sparse and I could be interpreting them incorrectly.

Regarding persistence, I don't think the trait cares whether it is persisted to disk. I think the `BasicQueue` is in-memory only. will update below to add that detail. I've also asked in riot to confirm.

Reply:
Okay I've addressed this by

1. rephrasing down to two uses of "import"
2. correcting the article to indicate that this is a regular old queue as you were expecting
3. adding details later that the `BasicQueue` is in-memory. (I'm still only mostly sure that this is correct @andresilva)